### DISCUSSION TITLE
Regarding Signing the AIR native extension

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#2453] [General]
As to achieve some custom requirement not available in AIR can be achieve by ANE (Android Native Extension).

Without signing the ANE, after building my custom ANE and packing with the my application, is it safe or it should be signin to secure the ANE.

@ajwfrost is there any free signing certificate available for ANE. 

Thanks.

Another issue was that, my Harman AIR license is over and while purchasing (not renewing as grace period is over) via Paypal, it's just loading not redirecting after entering my card details.

**Comment by ajwfrost** [comment]
Hi

Interestingly (and perhaps @marchbold can confirm whether this matches his experience), it looks like the ADT is all set up to be able to validate the signatures on ANE files - where they would need to be signed by signatures that can then be verified on the appropriate platform on which they're being used - but the setting is currently turned off by default...

So although it may be possible to sign an ANE using the normal signing inputs to ADT (which is a process similar to jarsigner for .jar and .aab files), it doesn't actually add anything or get checked at all....


For the Paypal issue -> I'm assuming you're not getting charged by Paypal for the attempts (!), can you email adobe.support and ask them to investigate that?

thanks

**Reply by marchbold** [reply]
We used to sign ANEs but found no advantages to it and haven't done it in over 10 years now. We didn't see anywhere that the signature was being checked so seemed a pointless exercise. 

I suppose if adt validated the signatures at package time or something it might be different, but we didn't find this to be the case and never got a straight answer from Adobe on what was happening.

**Reply by balaedit** [reply]
@ajwfrost  finally got loading issue resolved while purchasing Harman air SDK. I had purchased licence of Harman air today as grace period is over. Thanks
